.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,68 @@
+name: Build and Push
+
+# Builds and pushes the branch to ECR from master, or the manually selected branch when invoked manually. We stopped using Docker Hub for storing the image of oplogtoredis, where we used a Webhook to do the same thing.
+
+on:
+  push:
+    branches:
+      - master
+  # Left this here in-case the workflow needs to be developed further rapidly:
+  # pull_request:
+  #   types: [ opened, synchronize, reopened ]
+  workflow_dispatch:
+
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{ github.ref != 'refs/heads/master' }}
+
+permissions:
+  id-token: write # for the creds itself
+  contents: read # for checkout
+
+jobs:
+  build-and-push:
+    name: Build and Push
+    runs-on: ubuntu-latest
+    steps:
+
+      - uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608
+
+      - uses: cachix/install-nix-action@6ed004b9ccb68dbc28e7c85bee15fa93dbd214ac
+        with:
+          nix_path: nixpkgs=channel:nixos-unstable
+
+      # Examples of the generated tags:
+      # - v3.0.0 (branch: master)
+      # - v3.0.0-branch-name (branch: branch-name)
+      - name: Generate Tag
+        id: generate-tag
+        run: |
+          # Extract the current version from `default.nix`. This must pass.
+          version=$(nix flake show . --quiet --all-systems --json | jq -r '.defaultPackage."aarch64-darwin".name' | cut -d'-' -f2-)
```

Comment:
it's fine to hard-code `aarch64-darwin` ?

Reply:
I think it is, cause all the `${system}` variables will have the same `.name` property. Although I agree that it's ugly.